jc/parsers/iwconfig.py

```diff
@@ -0,0 +1,176 @@
+"""jc - JSON Convert `iwconfig` command output parser
+
+No `iwconfig` options are supported.
+
+Usage (cli):
+
+    $ iwconfig | jc --iwconfig
+
+or
+
+    $ jc iwconfig
+
+Usage (module):
+
+    import jc
+    result = jc.parse('iwconfig', iwconfig_command_output)
+
+Schema:
+
+    [
+    ]
+
+
+Examples:
+
+    $ iwconfig | jc --ifconfig -p
+
+"""
+import re
+from typing import List, Dict
+from jc.jc_types import JSONDictType
+import jc.utils
+
+class info():
+    """Provides parser metadata (version, author, etc.)"""
+    version = '1.0'
+    description = '`iwconfig` command parser'
+    author = 'Thomas Vincent'
+    author_email = 'vrince@gmail.com'
+    compatible = ['linux']
+    magic_commands = ['iwconfig']
+
+
+__version__ = info.version
+
+
+def _process(proc_data: List[JSONDictType]) -> List[JSONDictType]:
+    """
+    Final processing to conform to the schema.
+
+    Parameters:
+
+        proc_data:   (List of Dictionaries) raw structured data to process
+
+    Returns:
+
+        List of Dictionaries. Structured to conform to the schema.
+    """
+    int_list = ['signal_level', 'rx_invalid_nwid', 'rx_invalid_crypt', 'rx_invalid_frag', 
+        'tx_excessive_retries', 'invalid_misc', 'missed_beacon', 'tx_power', 'retry_short_limit']
+    float_list = ['frequency', 'bit_rate']
+    bool_list = ['rts_threshold', 'fragment_threshold', 'power_management']
+
+
+    proc_data = [ { key: int(value) if key in int_list else value for key, value in proc_data_item.items() } for proc_data_item in proc_data ]
+    proc_data = [ { key: float(value) if key in float_list else value for key, value in proc_data_item.items() } for proc_data_item in proc_data ]
+    proc_data = [ { key: value == 'on' if key in bool_list else value for key, value in proc_data_item .items() } for proc_data_item in proc_data ]
+
+    return proc_data
+
+def parse(
+    data: str,
+    raw: bool = False,
+    quiet: bool = False
+) -> List[JSONDictType]:
+    """
+    Main text parsing function
+
+    Parameters:
+
+        data:        (string)  text data to parse
+        raw:         (boolean) unprocessed output if True
+        quiet:       (boolean) suppress warning messages if True
+
+    Returns:
+
+        List of Dictionaries. Raw or processed structured data.
+    """
+    jc.utils.compatibility(__name__, info.compatible, quiet)
+    jc.utils.input_type_check(data)
+
+    raw_output: List[Dict] = []
+
+    # for backwards compatibility, preset all fields to None
+    wireless_extension_obj: Dict = {
```

Comment:
You can go ahead and make this a blank dictionary. This was just done for backwards compatibility with older versions of the `ifconfig` parser after I rewrote it.